staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,188 @@
+	w := &WhoAmIOptions{
+		ctx:       ctx,
+		IOStreams: flags.IOStreams,
+	}
+
+	clientConfig, err := flags.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientSet, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	w.authClient = clientSet.AuthenticationV1alpha1()
+
+	if !flags.PrintFlags.OutputFlagSpecified() {
+		w.printTable = true
+		w.printer = printers.NewTablePrinter(printers.PrintOptions{})
+	} else {
+		w.printer, err = flags.PrintFlags.ToPrinter()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return w, nil
+}
+
+// WhoAmIOptions is the start of the data required to perform the operation. As new fields are added,
+// add them here instead of referencing the cmd.Flags()
+type WhoAmIOptions struct {
+	authClient authenticationv1alpha1client.AuthenticationV1alpha1Interface
+	ctx        context.Context
+
+	printer    printers.ResourcePrinter
+	printTable bool
+
+	genericclioptions.IOStreams
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Experimental: Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl auth who-am-i
+		
+		# Get your subject attributes in JSON format.
+		kubectl auth who-am-i -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth who-am-i' sub command. Experimental.
+func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	flags := NewWhoAmIFlags(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "who-am-i",
+		DisableFlagsInUseLine: true,
+		Short:                 "Experimental: Check self subject attributes",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			o, err := flags.ToOptions(cmd.Context(), args)
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	flags.AddFlags(cmd)
+	return cmd
+}
+
+// Run prints all user attributes.
+func (o WhoAmIOptions) Run() error {
+	sar := &authenticationv1alpha1.SelfSubjectReview{}
+	response, err := o.authClient.SelfSubjectReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("The SelfSubjectReviews API is not enabled in the cluster.\n" +
+				"Enable APISelfSubjectReview featuee gate and authentication.k8s.io/v1alpha1 API.")
+		}
+		return err
+	}
+
+	if !o.printTable {
+		return o.printer.PrintObj(response, o.Out)
+	}
+
+	return o.printer.PrintObj(formatTable(response.Status.UserInfo), o.Out)
+}
+
+func formatTable(ui authenticationv1.UserInfo) *metav1.Table {
```

Comment:
Honestly, I'd consider creating your own printer for this custom type, rather than re-using TablePrinter, we already have places where we do so, see https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/apiresources/apiresources.go as one example. And use the built-in printers only for json|yaml output.

Reply:
I added a custom `ResourcePrinterFunc`.